omnigibson/utils/python_utils.py

```diff
@@ -726,3 +726,32 @@ def clear():
     """
     NAMES.clear()
     CLASS_NAMES.clear()
+
+
+def torch_delete(tensor, indices, dim=None):
+    """
+    Delete elements from a tensor along a specified dim.
+
+    Parameters:
+    tensor (torch.Tensor): Input tensor.
+    indices (int or array-like): Indices of elements to remove.
+    dim (int, optional): The dim along which to delete the elements.
+                          If None, the tensor is flattened before deletion.
+
+    Returns:
+    torch.Tensor: Tensor with specified elements removed.
+    """
+    if dim is None:
+        # Flatten the tensor if no dim is specified
+        tensor = tensor.flatten()
+        dim = 0
+
+    # Convert indices to a tensor if they are not already
+    if not isinstance(indices, th.Tensor):
+        indices = th.tensor(indices, dtype=th.long)
+
+    # Create a mask for the indices to keep
+    keep_indices = th.ones(tensor.size(dim), dtype=th.bool)
+    keep_indices[indices] = False
+
+    return tensor[keep_indices] if dim == 0 else tensor.transpose(0, dim)[keep_indices].transpose(0, dim)
```

Comment:
hmm the double `transpose` seems very inefficient? Also, are you assuming tensor is always a 1D or 2D tensor? If so, can we assert that first?